jsvg/src/main/java/com/github/weisj/jsvg/renderer/ShapeRenderer.java

```diff
@@ -108,29 +108,36 @@ public static void renderWithPaintOrder(@NotNull Graphics2D g, boolean canBeFill
         VectorEffect.applyEffects(shapePaintContext.vectorEffects, g,
                 shapePaintContext.context, shapePaintContext.transform);
 
+        Composite originalComposite = g.getComposite();
+        Paint originalPaint = g.getPaint();
+        Stroke originalStroke = g.getStroke();
+        AffineTransform originalTransform = g.getTransform();
+
```

Comment:
I think it would be worth while to extract this into a separate helper class. This way it is easier to also apply this in other situations.

Reply:
Thank you for the quick feedback, I've extracted it now.

I thought about making a proxy wrapper for a Graphics2D object that would track what changes are made to the original object, but I think that's overkill at this point, as there would be a lot of methods to override and keep track of.

Reply:
I agree that it would be a bit overkill. The current solution will be sufficient for most use cases 👍🏼